packages/flutter/lib/src/services/text_input.dart

```diff
@@ -856,6 +856,33 @@ abstract class TextInputClient {
   ///
   /// [TextInputClient] should cleanup its connection and finalize editing.
   void connectionClosed();
+
+  /// Requests that the client show the editing toolbar, for example when the
+  /// platform changes the selection through a non-flutter method such as
+  /// scribble.
+  void showToolbar();
+
+  /// UIKit has requested a text placeholder be added at the current selection
+  void insertTextPlaceholder(Size size);
+
+  /// UIKit has requested that the text placeholder be removed
+  void removeTextPlaceholder();
+}
+
+/// An interface for recieving focus during a UIIndirectScribbleInteraction
```

Comment:
Do you have any thoughts on what parts of this would be abstracted out? This is pretty closely tied to the way iPadOS's scribble feature interacts with the UITextInput, so I'm not sure how generic it would be

Reply:
I think I see what you mean – just keeping the docs at a more abstract level...I think I got all of those and made the docs more generic (e.g. handling x from engine) with the stuff about specific iOS features as a "for example"